bouncer/src/repo/core/model/bson/repo_bson_builder.h

```diff
@@ -69,10 +74,16 @@ namespace repo {
 					mongo::BSONObjBuilder::appendArray(label, bson);
 				}
 
+				
+				void appendMetadataVariant(
```

Comment:
Do we have purposes for the type outside of metadata that requires this exact combination of storage types?

If not, what is the benefit of reusing it? As potential downsides I see that it might take types that later purposes do not (or should not) allow and that any addition with another type later on for other purposes will mean that all visitors will need to be extended, even though they are not related to the purpose of the new type.

Admittedly, these are relatively small downsides, but making a new variant type for a new purpose is just another typedef definition, which is hardly much effort either.

Reply:
The intention is to use this  very same object type when we do #1 where we will no longer piggy-back off the mongo bson object

It wouldn't make sense to have such purpose built variant type and then setup something near identical again.

Reply:
Ah, I see. I will get to it then!

Reply:
Done. [3ef77d9](https://github.com/3drepo/3drepobouncer/pull/694/commits/3ef77d9fe126f47ef1c87449d029b1eaa19209fb)